server/statistics/hot_peer_cache.go

```diff
@@ -167,17 +167,39 @@ func (f *hotPeerCache) collectExpiredItems(region *core.RegionInfo) []*HotPeerSt
 // checkPeerFlow will update oldItem's rollingLoads into newItem, thus we should use write lock here.
 func (f *hotPeerCache) checkPeerFlow(peer *core.PeerInfo, region *core.RegionInfo) *HotPeerStat {
 	interval := peer.GetInterval()
-	// for test or simulator purpose
-	if Denoising && interval < HotRegionReportMinInterval {
+	if Denoising && interval < HotRegionReportMinInterval { // for test or simulator purpose
 		return nil
 	}
 	storeID := peer.GetStoreID()
 	deltaLoads := peer.GetLoads()
-	// update metrics
-	f.collectPeerMetrics(deltaLoads, interval)
+	f.collectPeerMetrics(deltaLoads, interval) // update metrics
 	regionID := region.GetID()
 	oldItem := f.getOldHotPeerStat(regionID, storeID)
 	thresholds := f.calcHotThresholds(storeID)
+
+	// check whether the peer is allowed to be inherited
+	source := direct
+	if oldItem == nil {
+		for _, storeID := range f.getAllStoreIDs(region) {
+			oldItem = f.getOldHotPeerStat(regionID, storeID)
+			if oldItem != nil && oldItem.allowInherited {
+				source = inherit
+				break
+			}
+		}
+	}
+
+	// check new item whether is hot
+	if oldItem == nil {
+		regionStats := f.kind.RegionStats()
+		isHot := slice.AnyOf(regionStats, func(i int) bool {
+			return deltaLoads[regionStats[i]]/float64(interval) >= thresholds[i]
+		})
+		if !isHot {
```

Comment:
why add this logic?

Reply:
I have brought forward the logic in updateNewHotPeerStat to avoid constructing an unnecessary hot peer stat